spec/layers.md

```diff
@@ -0,0 +1,81 @@
+clear function
+==============
+**Status**: explainer.
+
+Provide a `beginLayer()` and `endLayer()` functions that create and close layers for Canvas.
+
+
+Rationale
+---------
+
+Currently, the only way to draw a series of drawing calls at the same time, while applying the same sort of effects (shadows, blur, alpha, filters, compositing) to all of the drawn objects, without having these effects being applied one to one to this object, is by using an auxiliary canvas and then, drawing that auxiliary canvas to the canvas.
+This proposal adds an alternative to that (the same way that Fullter in Google and Core Graphics at Apple do) by creating a layer that then will be drawn in one go when the layer ends.
+
+
+Proposal
+--------
+
+```webidl
+interface CanvasRenderingContext2D {
```

Comment:
should this be part of `CanvasState`?

Reply:
I think these methods are part of the Context itself, the same way as save and restore.

Reply:
I think @fserb is right here, `save()` and `restore()` are defined on [CanvasState](https://html.spec.whatwg.org/multipage/canvas.html#canvasstate). This allows other interfaces like OffscreenCanvasRenderingContext2D or PaintWorklet's context to also inherit it, without having the full pack.

Reply:
Done!